en/developer_guide/best_practice_js.rst

```diff
@@ -0,0 +1,288 @@
+Best practices :
+================
+
+JavaScript is the main client-side scripting language used by many of open-source projects. This style guide is a list
+of dos and don'ts for JavaScript programs.
+
+Conventions :
+=============
+
+General Naming:
+---------------
+
+camelCase should be used for :
+
+- variable
+- function
+- object
+
+UPPERCASE_AND_UNDERSCORE should be used for :
+
+- constant
+
+PascalCase should be used for :
+
+- constructor
+- prototype
+- etc
+
+lowercase should be used for :
+
+- file name
+
+.. code-block:: js
+
+    functionNames;
+    variableNames;
+    ConstructorNames;
+    EnumNames;
+    methodNames;
+    SYMBOLIC_CONSTANTS;
+
+indentation:
+------------
+
+Operators must have a space before and after them.
+
+.. code-block:: js
+
+    var x = y % 2;
+
+A space must be added after each enumeration separator
+
+.. code-block:: js
+
+    var array = [“first”, “second”, “third”];
+
+Use a indention of 2 spaces to the blocks.
+
+.. code-block:: js
+
+    function () {
+      return true;
+    }
+
+Comments:
+---------
+
+The comments can use on one line to explain the code with `//`.
+Or use to create a advanced block with `/**/` to describe the functional, parameters, return, etc.. Also see `JsDoc`_, `GoogleDev`_
+and `Google Style Guide`_.
+
+.. code-block:: js
+
+    /** @const */ var MY_BEER = 'stout';
+
+    /**
+     * Determines whether a node is a field.
+     * @return {boolean} True if the contents of
+     *     the element are editable, but the element
+     *     itself is not.
+     * @deprecated Use isField().
+     */
+    BN_EditUtil.isTopEditableField = function(node) {
+      ...
+    };
+
+
+Variable declarations :
+-----------------------
+
+Declarations with var: Always
+
+When you fail to specify var, the variable gets placed in the global context, potentially altering existing values.
+Also, if there's no declaration, it's hard to tell in what scope a variable lives. So always declare with var.
+
+There is also the key word " `Let`_ " to declare a variable. The lifetime of this variable is that the block that declares
+it. Although promising it is not enough to be used implented.
+
+
+Constant declarations:
+----------------------
+
+Use `NAMES_LIKE_THIS` for constant values. Use `@const` to indicate, on comment, a constant
+(non-overwritable) pointer (a variable or property). Never use the " `const`_ " keyword as
+it's not supported in Internet Explorer before version 11 or not full supported.
```

Comment:
"Never" is a bit extreme. One may decide to not support IE and use const. Maybe change the sentence as an advice ?